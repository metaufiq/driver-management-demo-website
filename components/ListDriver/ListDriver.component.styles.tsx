import styled from "styled-components";

export const SpinnerContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 5rem;
`

export const ErrorContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  flex-direction: column;
  color: gray;
  gap: 1rem;
  padding: 4rem;
`

export const DataContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(5, minmax(0, 1fr));
  gap: 1rem;
  overflow-x: scroll;
  background-color: #fafafa;
  padding: 2rem 0;
  @media (max-width: 1430px) {
    grid-template-columns: repeat(4, minmax(0, 1fr));
  }
  @media (max-width: 1280px) {
    grid-template-columns: repeat(3, minmax(0, 1fr));
  }
  @media (max-width: 1024px) {
    grid-template-columns: repeat(2, minmax(0, 1fr));
  }
  @media (max-width: 768px) {
    grid-template-columns: repeat(1, minmax(0, 1fr));
  }
`