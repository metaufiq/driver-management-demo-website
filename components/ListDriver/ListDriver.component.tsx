import { Component } from "../../index.types"
import { PrimaryButton } from "../Button"
import DriverCard from "../DriverCard"
import Spinner from "../Spinner"
import { DataContainer, ErrorContainer, SpinnerContainer } from "./ListDriver.component.styles"
import { Props } from "./ListDriver.component.types"

const ListDriver:Component<Props> = ({isLoading, isError, data, refetch})=>{
  if (isLoading) {
   return(
    <SpinnerContainer>
      <Spinner size='4rem'/>
    </SpinnerContainer>
   ) 
  }

  if (isError) {
    return(
      <ErrorContainer>
        <p>Terjadi Kesalahan Saat pengambilan Data Driver.</p>
        <PrimaryButton onClick={refetch}>Coba Lagi</PrimaryButton>
      </ErrorContainer>
    )
  }

  return(
    <DataContainer>
      {data?.map(user=>(<DriverCard user={user} key={user.username}/>))}
    </DataContainer>
  )
}

export default ListDriver