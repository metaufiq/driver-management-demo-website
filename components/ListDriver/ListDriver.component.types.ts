import { Users } from "../../index.types"


export type Props = {
  data: Users,
  isLoading?: boolean,
  isError?: boolean,
  refetch?: VoidFunction
}