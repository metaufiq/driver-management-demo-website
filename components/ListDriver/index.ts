import ListDriver from "./ListDriver.component";

export default ListDriver