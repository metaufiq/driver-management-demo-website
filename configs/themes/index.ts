const themes = {
  colors: {
    primary: '#ff453b'
  },
};

export { themes };