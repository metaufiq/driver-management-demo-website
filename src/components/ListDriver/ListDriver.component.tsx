import { ReactElement } from "react"
import { Component } from "../../../index.types"
import Button from "../Button"
import DriverCard from "../DriverCard"
import Spinner from "../Spinner"
import { DataContainer, ErrorContainer, SpinnerContainer } from "./ListDriver.component.styles"
import { Props } from "./ListDriver.component.types"

/**
 * List Driver components
 * @param {Props} props component props 
 * @returns {ReactElement} List Driver components
 */
const ListDriver:Component<Props> = (
  {isLoading, isError, data, refetch}: Props
): ReactElement=>{
  if (isLoading) {
   return(
    <SpinnerContainer>
      <Spinner size='4rem'/>
    </SpinnerContainer>
   ) 
  }

  if (isError) {
    return(
      <ErrorContainer>
        <p>Terjadi Kesalahan Saat pengambilan Data Driver.</p>
        <Button onClick={refetch}>Coba Lagi</Button>
      </ErrorContainer>
    )
  }

  return(
    <DataContainer>
      {data?.map(user=>(<DriverCard user={user} key={user.username}/>))}
    </DataContainer>
  )
}

export default ListDriver