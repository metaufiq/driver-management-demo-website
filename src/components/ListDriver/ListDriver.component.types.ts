import { Users } from "@dmd/types"


export type Props = {
  data: Users,
  isLoading?: boolean,
  isError?: boolean,
  refetch?: VoidFunction
}