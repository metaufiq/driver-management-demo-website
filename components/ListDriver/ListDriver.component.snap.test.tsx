import { assertSnapshots } from "../../test";
import { Configs } from "../../test/index.types";
import ListDriver from ".";
import { Props } from "./ListDriver.component.types";
import { themes } from "../../configs/themes";

describe('ListDriverSnapshot', () => {

  const configs: Configs<Props> = [
    {
      desc: 'should render normal list driver',
      props:{
       data:[{
        dob: '12/10/1998',
        email: 'email@email.com',
        firstName: 'firstName',
        lastName: 'lastName',
        phone: '098123123123',
        username: 'userName',
        profilePicture: 'profile_picture_url'
       }], 
      },
      theme: themes,
      useTheme: true
    }
  ];

  assertSnapshots(ListDriver, configs);
});
