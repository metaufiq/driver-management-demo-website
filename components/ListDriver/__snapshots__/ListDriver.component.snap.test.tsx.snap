// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListDriverSnapshot should render normal list driver 1`] = `
<DocumentFragment>
  <div
    class="sc-ksZaOG gitNJv"
  >
    <div
      class="sc-eCYdqJ iywVtZ"
    >
      <div
        class="sc-gKXOVf bmManZ"
      >
        <div>
          <p
            class="sc-jSMfEi sc-iBkjds iqvdNZ hxEleX"
          >
            Driver ID
          </p>
          <p
            class="sc-ftvSup BYxNv"
          >
            userName
          </p>
        </div>
        <div
          class="sc-papXJ hdbybu"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 10a2 2 0 11-4 0 2 2 0 014 0zM12 10a2 2 0 11-4 0 2 2 0 014 0zM16 12a2 2 0 100-4 2 2 0 000 4z"
            />
          </svg>
        </div>
      </div>
      <img
        class="sc-hKMtZM iuUrIk"
        src="profile_picture_url"
      />
      <div
        class="sc-iqcoie fwuSZr"
      >
        <p
          class="sc-jSMfEi sc-jqUVSM iqvdNZ bwebnF"
        >
          Nama Driver
        </p>
        <p
          class="sc-kDDrLX gamHiL"
        >
          firstName, lastName
        </p>
      </div>
      <div
        class="sc-iqcoie fwuSZr"
      >
        <p
          class="sc-jSMfEi sc-jqUVSM iqvdNZ bwebnF"
        >
          Telepon
        </p>
        <p
          class="sc-kDDrLX gamHiL"
        >
          098123123123
        </p>
      </div>
      <div
        class="sc-iqcoie fwuSZr"
      >
        <p
          class="sc-jSMfEi sc-jqUVSM iqvdNZ bwebnF"
        >
          Email
        </p>
        <p
          class="sc-kDDrLX gamHiL"
        >
          email@email.com
        </p>
      </div>
      <div
        class="sc-iqcoie fwuSZr"
      >
        <p
          class="sc-jSMfEi sc-jqUVSM iqvdNZ bwebnF"
        >
          Tanggal Lahir
        </p>
        <p
          class="sc-kDDrLX gamHiL"
        >
          12/10/1998
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
